tests/cpptests/test_cpp_cursors.cpp

```diff
@@ -72,4 +74,41 @@ TEST_F(CursorsTest, OwnershipAPI) {
   ASSERT_EQ(Cursor_Free(cur), REDISMODULE_OK) << "Cursor should be deleted";
   ASSERT_EQ(Cursors_GetInfoStats().total_user, 0) << "Cursor should be deleted";
 
+  // Case 3: CursorList_Empty marks non-idle cursor for deletion
+  cur = Cursors_Reserve(&g_CursorsList, dummy, 1000, NULL);
+  ASSERT_TRUE(cur != NULL);
+  ASSERT_FALSE(cur->delete_mark);
+  ASSERT_FALSE(is_Idle(cur));
+  id = cur->id;
+
+  // Call CursorList_Empty while cursor is not idle (active)
+  CursorList_Empty(&g_CursorsList);
+
+  // Cursor should be marked for deletion, not immediately freed
+  ASSERT_EQ(Cursors_GetInfoStats().total_user, 1) << "Cursor should still be alive";
+  ASSERT_EQ(Cursors_TakeForExecution(&g_CursorsList, id), nullptr) << "Cursor already deleted";
+  ASSERT_TRUE(cur->delete_mark) << "Cursor should be marked for deletion";
+
+  // When cursor is paused, it should actually be freed due to delete_mark
+  ASSERT_EQ(Cursors_GetInfoStats().total_user, 1) << "Cursor should be alive";
+  ASSERT_EQ(Cursor_Pause(cur), REDISMODULE_OK) << "Pausing the cursor should actually free it";
+  ASSERT_EQ(Cursors_GetInfoStats().total_user, 0) << "Cursor should be deleted";
+
+  // Case 4: CursorList_Empty with explicit cursor free
```

Comment:
Consider adding a scenario with multiple cursors, some idle and some not, to test the `khash` iterator deleting while iterating, + keeping some of the iterators for concurrent release